File: matlab/timeConverter.m
function [ seconds ] = timeConverter( strTime )
% seconds = timeConverter( strTime )
% Converts HHMMSS input string to numeric seconds
%


len = length(strTime);

if len > 6
    seconds = 0;
else
    strTime = char(pad(strTime, 6, 'left', num2str(0)));

    h = str2num(strTime(1:2));
    m = str2num(strTime(3:4));
    s = str2num(strTime(5:6));

    seconds = s + m*60 + h*3600;
end
end

File: python/timeConverter.py
def timeConverter( strTime ):
	# int seconds = timeConverter( string strTime )
	# Converts HHMMSS input string to integer seconds
	#
	length = len(strTime)
	if length > 6:
		seconds = 0
	else:
		h = int(strTime[0:2])
		m = int(strTime[2:4])
		s = int(strTime[4:6])

		seconds = s + m*60 + h*3600
	return seconds
